pkg/controller/cyclenoderequest/transitioner/healthchecks.go

```diff
@@ -0,0 +1,222 @@
+package transitioner
+
+import (
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	v1 "github.com/atlassian-labs/cyclops/pkg/apis/atlassian/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// buildHealthCheckEndpoint generates the endpoint which will be used to perform a health check
+// It will render a string and replace {{ .NodeIP }} with the node private IP. If this is not present,
+// then the endpoint returned will be identical to the input
+func buildHealthCheckEndpoint(node corev1.Node, endpoint string) (string, error) {
+	tmpl, err := template.New("endpoint").Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
+
+	var privateIP string
+
+	// If there is no private IP, the error will be caught when trying
+	// to perform the health check on the node
+	for _, address := range node.Status.Addresses {
+		if address.Type == corev1.NodeInternalIP {
+			privateIP = address.Address
+		}
+	}
+
+	// Ensure other fields cannot be rendered to the filter
+	tmplStruct := struct {
+		NodeIP string
+	}{
+		NodeIP: privateIP,
+	}
+
+	var renderedEndpoint strings.Builder
+	if err = tmpl.Execute(&renderedEndpoint, tmplStruct); err != nil {
+		return "", err
+	}
+
+	return renderedEndpoint.String(), nil
+}
+
+// healthCheckPassed checks if the statusCode returned matches the set of valid status code for the health check
+// as well as if the body matches regex provided
+func healthCheckPassed(healthCheck v1.HealthCheck, statusCode uint, body []byte) (bool, string) {
+	// If there is not regex match string specified, don't check the body of the response
+	r := regexp.MustCompile(healthCheck.RegexMatch)
```

Comment:
MustCompile panics if it fails. User input should not panic the controller, it should be handled gracefully and fail the CNR if user feedback. 

Use regexp.Compile which returns an error